pkg/cmd/pr/checkout/checkout.go

```diff
@@ -204,7 +213,12 @@ func cmdsForMissingRemote(pr *api.PullRequest, baseURLOrName, repoHost, defaultB
 			// TODO: check if non-fast-forward and suggest to use `--force`
 			cmds = append(cmds, []string{"git", "fetch", baseURLOrName, fmt.Sprintf("%s:%s", ref, newBranchName)})
 		}
-		cmds = append(cmds, []string{"git", "checkout", newBranchName})
+
+		if opts.BranchName != "" {
+			cmds = append(cmds, []string{"git", "checkout", "-b", opts.BranchName, "--no-track", newBranchName})
```

Comment:
We should use `--track` instead of `--no-track` here, because `--track` is implied by `git checkout` if `--branch` wasn't specified (the `else` block in this conditional).

Reply:
hm. Regarding the new test, I'm not sure I know how I'd write that. Using what exists now, I thought I could just remove everything in `remotes` but the tests panic with what seems like an unrelated error. 

```go
		{
			name: "with local branch rename and no existing git remote",
			opts: &CheckoutOptions{
				SelectorArg: "123",
				BranchName:  "foobar",
				Finder: func() shared.PRFinder {
					baseRepo, pr := stubPR("OWNER/REPO:master", "hubot/REPO:feature")
					pr.MaintainerCanModify = true
					pr.HeadRepository = nil
					finder := shared.NewMockFinder("123", pr, baseRepo)
					return finder
				}(),
				Config: func() (config.Config, error) {
					return config.NewBlankConfig(), nil
				},
				Branch: func() (string, error) {
					return "main", nil
				},
			},
			remotes: map[string]string{},
			runStubs: func(cs *run.CommandStubber) {
				cs.Register(`git fetch origin refs/pull/123/head:feature`, 0, "")
				cs.Register(`git config branch\.feature\.merge`, 1, "")
				cs.Register(`git checkout -b foobar --no-track feature`, 0, "")
				cs.Register(`git config branch\.feature\.remote origin`, 0, "")
				cs.Register(`git config branch\.feature\.merge refs/pull/123/head`, 0, "")
			},
		},
```